flink-runtime/src/main/java/org/apache/flink/runtime/metrics/MetricRegistryConfiguration.java

```diff
@@ -43,8 +46,8 @@
 
 	private static volatile MetricRegistryConfiguration defaultConfiguration;
 
-	// regex pattern to split the defined reporters
-	private static final Pattern splitPattern = Pattern.compile("\\s*,\\s*");
+	// regex pattern to extract the name from reporter configuration keys, e.g. "rep" from "metrics.reporter.rep.class"
+	private static final Pattern configurationPattern = Pattern.compile("metrics\\.reporter\\.([\\S&&[^.]]*)\\..*");
```

Comment:
Can we `Pattern.quote(ConfigConstants.METRICS_REPORTER_PREFIX)`?

Reply:
yes